edgexfoundry/go-mod-bootstrap
# refactor!: Replace SecretStore service config with default values and overrides

BREAKING CHANGE: SecretStore config no longer in service configuration file. Changes must be done via use of environment variable overrides of default values

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  TBD

## Testing Instructions
Build core data with this branch
Run secure EdgeX stack, store core data container and remove core data config from Consul
Run core data locally with DEBUG logging and SecretStore enabled.
```
sudo EDGEX_SECURITY_SECRET_STORE=true WRITABLE_LOGLEVEL=DEBUG ./core-data -cp -r
```
Verify core data bootstraps successfully and is receiving event from MessageBus
Verify core data config loaded from file and pushed to Consul 
Verify core data config in Consul no longer contains SecretStore section
Restart core data
Verify core data config loaded from Consul, never from file. 

More extensive testing will occur when edgex-go consumes these changes.

## New Dependency Instructions (If applicable)
